azext_iot/tests/iothub/devices/test_iothub_device_tracing.py

```diff
@@ -9,10 +9,11 @@
 from azext_iot.tests import IoTLiveScenarioTest
 from azext_iot.tests.settings import DynamoSettings, ENV_SET_TEST_IOTHUB_BASIC
 from azext_iot.common.shared import AuthenticationTypeDataplane
+from azext_iot.tests.generators import generate_generic_id
 
 settings = DynamoSettings(req_env_set=ENV_SET_TEST_IOTHUB_BASIC)
 
-LIVE_HUB = settings.env.azext_iot_testhub
+LIVE_HUB = settings.env.azext_iot_testhub if settings.env.azext_iot_testhub else "test-hub-" + generate_generic_id()
```

Comment:
Instead of repeating this code for iot hub name generation across the IoT Hub IT modules, we could apply a similar pattern once (reduce code duplication) in the base test class like `IoTLiveScenarioTest` for IoT Hub ITs. First `azext_iot_testhub` must be an optional parameter with respect to its usage with `DynamoSettings`. Then during instance initialization the logic can check if an entity name has been set, if not generate one.  If generated, then create the hub. Similarly during teardown/cleanup process only delete the hub if its generated.